jsonschema/_utils.py

```diff
@@ -158,10 +158,46 @@ def ensure_list(thing):
     return thing
 
 
+def dict_equal(one, two):
+    """
+    Check if two dicts are the same using `equal`
+    """
+    if len(one.keys()) != len(two.keys()):
+        return False
+
+    for key in one:
+        if key not in two:
+            return False
+        if not equal(one[key], two[key]):
+            return False
+
+    return True
+
+
+def list_equal(one, two):
+    """
+    Check if two lists are the same using `equal`
+    """
+    if len(one) != len(two):
+        return False
+
+    for i in range(0, len(one)):
+        if not equal(one[i], two[i]):
+            return False
+
+    return True
+
+
 def equal(one, two):
     """
     Check if two things are equal, but evade booleans and ints being equal.
     """
+    if isinstance(one, list) and isinstance(two, list):
```

Comment:
@Julian should I also extract `list_equal` and `dict_equal` into a dedicated PR for #686?

Reply:
Yes please! Especially because this implementation will need some tweaking, it will break for arbitrary mappings and sequences, which we support (so we need to add tests for that).

But let's take that to a separate PR, appreciated!

Reply:
Moved to https://github.com/Julian/jsonschema/pull/820. Test cases will be currently ignored in draft2020-12, once merged I can rebase an remove the commit.